### PR TITLE
Create Rugpull Attacks

### PR DESCRIPTION
**Pull request by Gul-Hameed007**
Title: [Rugpull] Kokomo Finance Exploit

This pull request adds a detailed description of a new vulnerability, specifically a rugpull incident involving Kokomo Finance that occurred on March 27, 2023. The exploit led to the loss of approximately $4 million in Wrapped Bitcoin (WBTC).

Rugpulls represent a significant risk in the cryptocurrency and DeFi ecosystem. They typically involve developers or project owners suddenly ceasing operations and making off with user funds. This unexpected action often leaves users with no recourse and results in significant financial losses.

In the case of Kokomo Finance, the deployer address made a malicious modification to the smart contract, redirecting WBTC deposits to their own address. Shortly after this operation, the project's online presence, including its website, Twitter, GitHub, and Medium accounts, was deleted.

I have provided a step-by-step reproduction of the exploit, a detailed transaction analysis, and a list of red flags that could signal a potential rugpull.

This rugpull exploit is categorized according to the STRIDE and MITRE ATT&CK frameworks as follows:

STRIDE Category: Tampering
MITRE ATT&CK Tactic: Impact
MITRE ATT&CK Technique: Endpoint Denial of Service (T1499) Preventing rugpulls can be challenging due to the decentralized nature of DeFi. In light of this, I have proposed several mitigation measures, including regular audits, identifying a reputable team, implementing lockup periods, decentralizing control, and encouraging transparency in DeFi projects.

I hope that the addition of this vulnerability and its mitigation methods will aid in increasing awareness and improving security within the DeFi ecosystem.

Source: https://rekt.news/kokomo-finance-rekt/

### CODE REVIEW

File: oswar/persistence/Rugpull
Comment (dmkulazhenko) [inline oswar/persistence/Rugpull:1]: files describing attacks should use Markdown markup language features, so... at least headings should be marked.

Here you can learn more about Markdown: https://www.markdownguide.org/basic-syntax/

**Review by dmkulazhenko — CHANGES_REQUESTED** [review]
Hi @Gul-Hameed007!
Happy to see you as contributor to OSWAR, really appreciate it!
-----

https://github.com/CyVers-AI/oswar/blob/main/mapping.json
mapping file should be updated to render new file (attack).

File: oswar/persistence/Rugpull
[the remark this answers is not in the document]
Comment (Gul-Hameed007) [reply oswar/persistence/Rugpull:1]: Done

Comment (Gul-Hameed007) [reply oswar/persistence/Rugpull:1]: Done

**Comment by dmkulazhenko** [comment]
@Gul-Hameed007, move changes from #8 to this PR.

**Comment by Gul-Hameed007** [comment]
@dmkulazhenko i've updated the mapping.json file in this PR.

**Review by dmkulazhenko — APPROVED** [review]
lgtm

**Review by Raiders0786 — APPROVED** [review]
Thank you for contributing @Gul-Hameed007 🚀

**Pull request merged**